File: src/main/java/com/ceshiren/appcrawler/DemoPlugin.java
package com.ceshiren.appcrawler;

import com.ceshiren.appcrawler.core.Crawler;
import com.ceshiren.appcrawler.model.URIElement;
import com.ceshiren.appcrawler.plugin.Plugin;

public class DemoPlugin extends Plugin {
    @Override
    public void init(Crawler crawler) {
        System.out.println("demo init");
        log().debug("demo init");
    }

    @Override
    public void beforeElementAction(URIElement element) {
        log().info(element);

    }
}

File: src/main/scala/com/ceshiren/appcrawler/plugin/DemoPlugin.scala
package com.ceshiren.appcrawler.plugin

import com.ceshiren.appcrawler.model.URIElement

/**
  * Created by seveniruby on 16/1/21.
  */
class DemoPlugin extends Plugin{
  override def beforeElementAction(element: URIElement): Unit ={
    log.info("demo com.ceshiren.appcrawler.plugin before element action")
    log.info(element)
    log.info("demo com.ceshiren.appcrawler.plugin end")
  }
  override def afterUrlRefresh(url:String): Unit ={
    getCrawler().currentUrl=url.split('|').last
    log.info(s"new url=${getCrawler().currentUrl}")
    if(getCrawler().currentUrl.contains("Browser")){
      getCrawler().getBackButton()
    }
  }

}
